File: fixtures/polyglot/go/orders/payment.go
package orders

import "billing/invoice"

type PaymentService struct {
  Repo invoice.Repository
  CurrentUser string
}

type PaymentPort interface {
  Charge(amount int) Receipt
}

func (p *PaymentService) Charge(amount int) Receipt {
  return Receipt{}
}

type Receipt struct {
  Amount int
}

File: fixtures/polyglot/javascript/orders/payment.js
const { InvoiceRepository } = require("../billing/invoice");

class PaymentService {
  constructor(invoices = new InvoiceRepository()) {
    this.invoices = invoices;
  }

  charge(request) {
    return this.invoices.create(request);
  }
}

module.exports = { PaymentService };
